pkg/network/render.go

```diff
@@ -615,3 +629,23 @@ func renderNetworkDiagnostics(conf *operv1.NetworkSpec, manifestDir string) ([]*
 
 	return manifests, nil
 }
+
+// renderCloudNetworkConfigController renders the cloud network config controller
+func renderCloudNetworkConfigController(conf *operv1.NetworkSpec, cloudBootstrapResult bootstrap.CloudBootstrapResult, manifestDir string) ([]*uns.Unstructured, error) {
+	data := render.MakeRenderData()
+	data.Data["ReleaseVersion"] = os.Getenv("RELEASE_VERSION")
+	data.Data["CloudProvider"] = cloudBootstrapResult.Platform
+	data.Data["CloudRegion"] = cloudBootstrapResult.Region
+	data.Data["CloudProviderAWS"] = v1.AWSPlatformType
+	data.Data["CloudProviderAzure"] = v1.AzurePlatformType
+	data.Data["CloudProviderGCP"] = v1.GCPPlatformType
```

Comment:
question CloudProvider should be sufficient to capture different public cloud types so I am not sure why we need the other fields CloudProviderAWS/CloudProviderAzure/CloudProviderGCP , seems like redundancy to me no ?

Reply:
Not really: the cloud-network-config-controller needs credentials to be able to talk to the cloud API, the component generating the credentials is the cloud-credentials-operator. That component watches a CRD for which the CNO will write a CR, defined here: https://github.com/openshift/cluster-network-operator/blob/5617e47380541826e9496af905c2b0e3effba70a/bindata/cloud-network-config-controller/005-credentialrequest.yaml . As you can see the spec for that CR changes depending on the cloud provider, so we will need to determine which cloud we're deploying in as to write the correct CR, these are just helper variables to make that determination.